doc/lua_api.md

```diff
@@ -7443,6 +7445,20 @@ use the provided load and write functions for this.
 * `from_file(filename)`: Experimental. Like `from_string()`, but reads the data
   from a file.
 
+`AsyncJob`
+----------
+An `AsyncJob` is a reference to a job to be run in an async environment.
+
+### Methods
+* `get_id()`: get the numeric ID of the job.
+* `replace(func, callback, ...)`:
+    * Try to replace the job or (if not possible) queue a new job.
+    * Note that the `func`, `callback` and variable arguments are all replaced.
```

Comment:
I am not sure. The current situation is that the job queue and the IDs are both continuous, so finding the a job in the queue can be easily implemented by computing the index into the job queue. Canceling a job would break this and require manually iterating through the queue and removing the job.

Reply:
You could avoid that by simply setting a cancelled flag on the job structure.
But I think iteration doesn't matter either. In practice the queue should be quite short. Could also use a binary search if that's an issue.

> The other thing is that replacing the job in-place would theoretically allow the job to be executed earlier, so for situations involving (1) multiple time-consuming jobs [...]

That's true. I think this doesn't matter in practice for the same reason.
Depending on how mods use the async jobs there could be noticeable latency if someone fills the queue with time-consuming jobs (and the max threads are reached). So it's better if people don't.

Reply:
> You could avoid that by simply setting a cancelled flag on the job structure.

I went for this approach since this is (IMO) the most straightforward one.